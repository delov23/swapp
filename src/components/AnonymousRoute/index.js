export { default as AnonymousRoute } from './AnonymousRoute';
