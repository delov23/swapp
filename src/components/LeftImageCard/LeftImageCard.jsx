import React, { useContext } from 'react';
import { Link } from 'react-router-dom';

import styles from './LeftImageCard.module.css';
import ThemeContext from '../../context/ThemeContext';
import { themify } from '../../utils/themify';

const LeftImageCard = ({ width, image, text, href }) => {
  const theme = useContext(ThemeContext);

  return (
    <Link
      to={href}
      style={{ width }}
      className={styles[themify('card', theme)]}
    >
      <img className={styles.image} src={image} alt="" />
      <h3 className={styles.heading}>{text}</h3>
    </Link>
  );
};

export default LeftImageCard;
