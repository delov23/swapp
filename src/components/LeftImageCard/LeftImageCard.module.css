.card {
  width: 28vw;
  display: flex;
  justify-content: flex-start;
  margin: 5px;
  height: 5rem;
  align-items: center;
  border-width: 1px;
  border-style: solid;
  font-family: Starwars;
  border-radius: 10px;
  -webkit-border-radius: 10px;
  -moz-border-radius: 10px;
  -ms-border-radius: 10px;
  -o-border-radius: 10px;
}

.cardLight {
  composes: card;
  background-color: #fff;
  color: var(--mainCyan);
  border-color: #e5e9f2;
}

.cardDark {
  composes: card;
  background-color: #333;
  color: var(--mainYellow);
  border-color: #3c4858;
}

.image {
  height: 5rem;
  border-top-left-radius: 10px;
  border-bottom-left-radius: 10px;
}

.heading {
  font-family: Starwars;
  font-weight: 100;
  margin: auto;
  text-align: center;
}

@media screen and (max-width: 850px) {
  .card {
    width: 100% !important;
  }
}