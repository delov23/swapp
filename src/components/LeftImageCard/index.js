export { default as LeftImageCard } from './LeftImageCard';
