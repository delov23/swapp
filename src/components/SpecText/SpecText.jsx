import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import ThemeContext from '../../context/ThemeContext';
import { themify } from '../../utils/themify';
import styles from './SpecText.module.css';

const SpecText = ({ left, right }) => {
  const theme = useContext(ThemeContext);

  return (
    <p className={styles.text}>
      <span className={styles[themify('left', theme)]}>{left}: </span>
      <span className={styles[themify('right', theme)]}>{right}</span>
    </p>
  );
};

SpecText.propTypes = {
  left: PropTypes.string.isRequired,
  right: PropTypes.string.isRequired,
};

export default SpecText;
