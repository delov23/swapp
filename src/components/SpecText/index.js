export { default as SpecText } from './SpecText';
