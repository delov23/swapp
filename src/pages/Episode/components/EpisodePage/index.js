export { default as EpisodePage } from './EpisodePage';
