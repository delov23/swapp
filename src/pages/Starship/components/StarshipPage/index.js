export { default as StarshipPage } from './StarshipPage';
