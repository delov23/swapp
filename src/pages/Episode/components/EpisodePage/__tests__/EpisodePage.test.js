import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import { EpisodePage } from '../';
import { LeftImageCard } from '../../../../../components/LeftImageCard';
import { SpecText } from '../../../../../components/SpecText';

const loadMock = jest.fn();

const EXAMPLE_PROPS = {
  loadMore: new loadMock(),
  people: {
    edges: [
      {
        node: {
          id: 'person.1',
          name: 'Name 11',
          image: 'image 2',
        },
        cursor: 'HASH_1',
      },
      {
        node: {
          id: 'person.2',
          name: 'Name 22',
          image: 'image 3',
        },
        cursor: 'HASH_2',
      },
    ],
    pageInfo: {
      hasNextPage: false,
      endCursor: 'HASH_2',
    },
  },
  episode: {
    id: 'episode.1',
    title: '',
    openingCrawl: 'Long......',
    director: 'Name',
    releaseDate: '1993-11-02',
    episodeId: 10,
    image: 'image',
  },
  loadingCharacters: false,
};

describe('<EpisodePage />', () => {
  it('should match snapshot', () => {
    expect(shallow(<EpisodePage {...EXAMPLE_PROPS} />)).toMatchSnapshot();
  });

  it('should display information', () => {
    const wrapper = mount(
      <Router>
        <EpisodePage {...EXAMPLE_PROPS} />
      </Router>,
    );

    expect(wrapper.find(LeftImageCard)).toHaveLength(2);
    expect(wrapper.find(SpecText)).toHaveLength(2);
  });
});
