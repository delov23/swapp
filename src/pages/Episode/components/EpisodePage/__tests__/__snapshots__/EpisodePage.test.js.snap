// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EpisodePage /> should match snapshot 1`] = `
<main>
  <section
    className="posterLight"
  >
    <img
      alt="Episode"
      className="image"
      src="image"
    />
    <div
      className="titles"
    >
      <h1
        className="titleLight"
      >
        Star Wars: Episode 
        10
      </h1>
      <h2
        className="subtitleLight"
      />
    </div>
  </section>
  <div
    className="boxLight"
  >
    <p
      className="opening"
    >
      Long......
    </p>
    <SpecText
      left="Director"
      right="Name"
    />
    <SpecText
      left="Release Date"
      right="1993-11-02"
    />
  </div>
  <section>
    <div
      style={
        Object {
          "display": "flex",
          "flexWrap": "wrap",
        }
      }
    >
      <LeftImageCard
        href="/characters/person.1"
        image="image 2"
        key="person.1"
        text="Name 11"
        width="30%"
      />
      <LeftImageCard
        href="/characters/person.2"
        image="image 3"
        key="person.2"
        text="Name 22"
        width="30%"
      />
    </div>
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    />
  </section>
</main>
`;
